pkg/services/pluginsintegration/pluginconfig/request.go

```diff
@@ -95,6 +96,14 @@ func (s *RequestConfigProvider) PluginRequestConfig(ctx context.Context, pluginI
 			m[azsettings.AzureCloud] = azureSettings.Cloud
 		}
 
+		if customClouds := azureSettings.CustomCloudList; len(customClouds) > 0 {
+			if jsonBytes, err := json.Marshal(customClouds); err != nil {
```

Comment:
I'm not sure on this and the impact it may have on each plugin request. cc @wbrowne

Reply:
Yeah, ideally we would cache this serialized `CustomCloudList` once and send it to the plugin process once.  Maybe there is a better place to do this?

Reply:
@JonCole If possible we tend to avoid relying on the (now legacy) environment variable approach for security reasons. Adding them to the request (as you have here) is the recommended way to go. If we have some way of grabbing the JSON right off the config ini and keeping it somewhere (`azureSettings` perhaps) so it can be passed here directly rather than marshalled every time, then that sounds good to me.

Reply:
@wbrowne oh, I think I misunderstood - you are saying you want me to avoid the json serialization by caching the json payload locally then we still set that json string into the map?